endpoint.go

```diff
@@ -767,13 +767,15 @@ func (ep *endpoint) assignAddress(assignIPv4, assignIPv6 bool) error {
 	}
 
 	if assignIPv4 {
-		if err = ep.assignAddressVersion(4, ipam); err != nil {
+		if err = ep.assignAddressVersion(4, ipam); err != nil && !ep.network.enableIPv6 {
```

Comment:
I think this change is needed only to get the things work in case we overload `enableIPv6` with the `allocate ipv6 only` message.